packages/web3-eth-contract/src/index.js

```diff
@@ -123,8 +123,16 @@ var Contract = function Contract(jsonInterface, address, options) {
 
 
                     // add method only if not one already exists
-                    if(!_this.methods[method.name])
+                    if(!_this.methods[method.name]) {
                         _this.methods[method.name] = func;
+                    } else {
+                        var cascadeFunc = _this._createTxObject.bind({
+                            method: method,
+                            parent: _this,
+                            next: _this.methods[method.name]
```

Comment:
Could we please rename this to: `next` -> `nextMethod`

btw we have the same issue for events :0